### PR TITLE
Add `python3Packages.ckcc-protocol` package, and list it as a dependency for `electrum`.

### PR DESCRIPTION
**Pull request by hkjn**
###### Motivation for this change

The https://coldcardwallet.com/ is a hardware wallet for Bitcoin. This change adds the `python3Packages.ckcc-protocol` package, and lists it as a dependency for `electrum`.

Similar to `keepkey`, `trezor` and `btchip`, `ckcc-protocol` is a plugin,
allowing electrum to communicate with the https://coldcardwallet.com/.

Also ~add `hkjn` to maintainers, and~ list `hkjn` as initial maintainer for `ckcc-protocol`. (Would be happy to hand over to official Coldcard maintainers at any point, if they are willing.)

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

N/A

### CODE REVIEW

**Comment by hkjn** [comment]
cc @mmahut @worldofpeace.

Sorry for semi-randomly pulling both of you into this PR, but I wanted to check for your advice on how to proceed with this ~1.5 months old PR to add a new package and list it as a dependency for `electrum`, to be able to use Coldcard devices with Electrum. I haven't received a review so far from @FRidh, who was auto-added as code owner.

Thanks in advance!

**Review by jb55 — APPROVED** [review]

**Comment by mmahut** [comment]
I'm sorry for missing this mention.

@GrahamcOfBorg build pythonPackages.ckcc-protocol python3Packages.ckcc-protocol

**Comment by mmahut** [comment]
Looks like ckcc-protocol requires python 3.6+, please use something like ```disabled = ! isPy3k;``` in the package.

**Comment by hkjn** [comment]
Thanks for the review @mmahut and @jb55, please take another look!

> Looks like ckcc-protocol requires python 3.6+, please use something like disabled = ! isPy3k; in the package.

I added the following to ckcc-protocol package, using other Python packages as a reference:

```
disabled = ! pythonAtLeast "3.6";
```

I've also bumped the package version to 0.8.0, which was published between the time I sent this PR out and at time of writing.

File: pkgs/development/python-modules/ckcc-protocol/default.nix [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1mbs9l8qycy50j5lq6az7l5d8i40nb0vmlyhcyax298qp6c1r1gh";
+  };
+
+  checkInputs = [
+    pytest
+  ];
+
+  disabled = ! pythonAtLeast "3.6";
```
Comment (jonringer) [inline pkgs/development/python-modules/ckcc-protocol/default.nix]: i would also put this below version
```suggestion
  disabled = pythonOlder "3.6";
```

File: pkgs/development/python-modules/ckcc-protocol/default.nix [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+  propagatedBuildInputs = [ click ecdsa hidapi pyaes ];
+
+  meta = {
```
Comment (jonringer) [inline pkgs/development/python-modules/ckcc-protocol/default.nix]:
```suggestion
  meta = with lib; {
```

File: pkgs/development/python-modules/ckcc-protocol/default.nix [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    description = "Communicate with your Coldcard using Python";
+    homepage = https://github.com/Coldcard/ckcc-protocol;
+    license = lib.licenses.gpl3;
```
Comment (jonringer) [inline pkgs/development/python-modules/ckcc-protocol/default.nix]:
```suggestion
    license = licenses.gpl3;
```

File: pkgs/development/python-modules/ckcc-protocol/default.nix [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    maintainers = with lib.maintainers; [ hkjn ];
```
Comment (jonringer) [inline pkgs/development/python-modules/ckcc-protocol/default.nix]:
```suggestion
    maintainers = with maintainers; [ hkjn ];
```

File: pkgs/development/python-modules/ckcc-protocol/default.nix (hunk 1 above)
Comment (hkjn) [reply pkgs/development/python-modules/ckcc-protocol/default.nix]: Thanks, good suggestion. Done.

File: pkgs/development/python-modules/ckcc-protocol/default.nix (hunk 3 above)
Comment (hkjn) [reply pkgs/development/python-modules/ckcc-protocol/default.nix]: Done.

File: pkgs/development/python-modules/ckcc-protocol/default.nix (hunk 4 above)
Comment (hkjn) [reply pkgs/development/python-modules/ckcc-protocol/default.nix]: Done.

File: pkgs/development/python-modules/ckcc-protocol/default.nix (hunk 2 above)
Comment (hkjn) [reply pkgs/development/python-modules/ckcc-protocol/default.nix]: Done.

**Comment by hkjn** [comment]
Thanks for the review @jonringer. I've addressed your suggestions at this time, please take another look.

**Comment by mmahut** [comment]
@GrahamcOfBorg build python3Packages.ckcc-protocol electrum

**Comment by hkjn** [comment]
Ping @jonringer @mmahut @jb55. Any chance for a review so this ~4 month old PR can move towards merging?

Thanks in advance!

**Comment by jonringer** [comment]
the rebase, "Showing 4,569 changed files with 111,966 additions and 116,126 deletions.". That's a lot of changes in a few months O.o

**Comment by hkjn** [comment]
> the rebase, "Showing 4,569 changed files with 111,966 additions and 116,126 deletions.". That's a lot of changes in a few months O.o

Agreed, although with 200k+ commits in the repo, I guess this is to be expected? https://github.com/NixOS/nixpkgs/commits/master

Or did I misunderstand, and there's some issue with the two commits I'd like to merge here, or the way I'm rebasing on top of recent commits in upstream's master?

**Comment by jonringer** [comment]
you're good, it's just the force push "brings along all the changes" in the force-push view. Your PR looks good though, reviewing now

**Review by jonringer — APPROVED** [review]
diff LGTM
commits LGTM
ckcc shows usage
electrum starts up just fine

only a very small difference in closure size:
```
$ nix path-info -Sh ./result
/nix/store/if5bzxc9ls6wmrql4fqkfsq3hbv0h1d8-electrum-3.3.8	 654.4M
[nix-shell:/home/jon/.cache/nix-review/pr-66516]$ nix path-info -Sh ./results/electrum
/nix/store/nmc3fraynkghqjm0cysgf38l9f50yz4m-electrum-3.3.8	 654.5M
```
```
[5 built, 29 copied (17.1 MiB), 10.0 MiB DL]
https://github.com/NixOS/nixpkgs/pull/66516
3 package were built:
electrum python37Packages.ckcc-protocol python38Packages.ckcc-protocol
```

**Comment by jonringer** [comment]
@GrahamcOfBorg build electrum python37Packages.ckcc-protocol python38Packages.ckcc-protocol

**Pull request merged**